python/cudf/cudf/core/multiindex.py

```diff
@@ -762,6 +763,20 @@ def _compute_validity_mask(self, index, row_tuple, max_length):
             if isinstance(row, slice) and row == slice(None):
                 continue
             lookup[i] = cudf.Series(row)
```

Comment:
out of scope of this PR but it would probably improve perf here if we didn't go through the whole song and dance of constructing a series for each column we're setting in and furthermore going through `DataFrame.__setitem__` each time

Reply:
All of this dataframe/series level code everywhere should be rewritten one level down operating directly on columns, but ...